addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -977,9 +977,21 @@ const Wysiwyg = Widget.extend({
                         name: el.dataset.fileName || '',
                         data: el.getAttribute('src').split(',')[1],
                         is_image: true,
+                        ...(this.options.imageAsPublicAttachment ? {} : this.options.recordInfo),
                     },
                 });
-                el.setAttribute('src', attachment.image_src);
+                let src = attachment.image_src;
+                // TODO on master: make this a single rpc call, with batched
+                // attachment creation and access token generation.
+                if (!attachment.public) {
+                    const [accessToken] = await this._rpc({
+                        model: 'ir.attachment',
+                        method: 'generate_access_token',
+                        args: [attachment.id],
+                    });
+                    src += `?access_token=${encodeURIComponent(accessToken)}`;
+                }
```

Comment:
As explained in the comment of the previous commit, I think we might not need this after all if we disable the automatic b64 conversion to attachment for models that have been specified as requiring an inline conversion. I don't remember the name of the option from the top of my hat but it's something like `inline-style: true`.

Reply:
The `inlineStyle` options is true for mail.compose.message.
But not for mailing.mailing.